src/plugins/kibana_legacy/server/index.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { CoreSetup, CoreStart, PluginConfigDescriptor } from 'kibana/server';
+
+import { configSchema, ConfigSchema } from '../config';
+
+export const config: PluginConfigDescriptor<ConfigSchema> = {
+  exposeToBrowser: {
+    defaultAppId: true,
+  },
+  schema: configSchema,
+  deprecations: ({ renameFromRoot }) => [
+    // TODO: Remove deprecation once defaultAppId is deleted
+    renameFromRoot('kibana.defaultAppId', 'kibana_legacy.defaultAppId'),
```

Comment:
This triggers a warning in the server logs when `kibana.defaultAppId` is used:
```
  log   [16:23:34.047] [warning][config][deprecation] "kibana.defaultAppId" is deprecated and has been replaced by "kibana_legacy.defaultAppId"
```

Since we are going to sunset the whole config key soon it doesn't really help when users are prompted to change the setting now.

It seems like an easy way to do this is to no pass down the logger function to the helper:
```ts
[
    // TODO: Remove deprecation once defaultAppId is deleted
    (config, rootPath) => renameFromRoot('kibana.defaultAppId', 'kibana_legacy.defaultAppId')(config, rootPath, /* don't log this deprecation because it won't stay long*/ () => {}),
```

Reply:
I was suggesting to copy the `renameFromRoot` code to ignore the logger, as this was temporary. but the suggestion may be better